packages/rabbitmq/src/amqp/connection.ts

```diff
@@ -429,4 +407,31 @@ export class AmqpConnection {
 
     return handler(message, msg);
   }
+
+  private async setupQueue<T>(
+    subscriptionOptions: MessageHandlerOptions,
+    channel: amqplib.ConfirmChannel
+  ) {
+    const { exchange, routingKey } = subscriptionOptions;
+    let queue;
+    if (subscriptionOptions.createQueueWhenNotExisting) {
+      queue = await channel.assertQueue(
+        subscriptionOptions.queue || '',
+        subscriptionOptions.queueOptions || undefined
+      );
+    } else {
+      queue = await channel.checkQueue(subscriptionOptions.queue || '');
+    }
+    const routingKeys = Array.isArray(routingKey) ? routingKey : [routingKey];
+    if (exchange && routingKeys) {
```

Comment:
I think this condition might be wrong. You'll still need to bind the queue even if no specific exchange was provided but it should fall back to the default exchange. Otherwise, you won't be able to receive any messages

Reply:
Thx, I just got the tests running so I will check & fix that.

Reply:
At least all tests pass. On another note: Can you hint me to where I need to change the decorator `RabbitSubscribe`? I cannot find the code where exchange and routing key are made required (in `MessageHandlerOptions` they are all optional).

Reply:
We will definitely need to add new e2e tests to verify that this all works as intended. Given the current code I really don't think you'll receive any messages since no binding is being created. You need to actually bind the queue, doing `check` or `assert` won't actually attach your handler.

Not sure what you mean about `RabbitSubscribe`, the changes to `MessageHandlerOptions` should be all that is required. Maybe you need to rebuild the project if you're getting Typescript errors? You can just use `yarn build:watch` if you want to have everything kept up to date automatically as you make changes

Reply:
When I use the `@RabbitSubscribe` decorator (as done in all the tests), `exchange`, `routingKey` and `queue` are required. I cannot find the code that makes them optional. I believe it may be this snippet, but I am not good enough with TypeScript to fully understand what that code does (coming from a strongly typed proprietary language, so all this is pretty new for me)

```javascript
export const makeRabbitDecorator = <T extends Partial<RabbitHandlerConfig>>(
  input: T
) => (
  config: Pick<RabbitHandlerConfig, Exclude<keyof RabbitHandlerConfig, keyof T>>
) => (target, key, descriptor) =>
  SetMetadata(RABBIT_HANDLER, { ...input, ...config })(target, key, descriptor);
```

I will add new e2e tests to verify original behavior and to cover the new behavior. It just takes more time, because I am learning all this on the fly :D.

Thx for your patience and support, btw!

Reply:
@j-schreiber The code you've shown here is not really related. The only necessary change is to mark the properties in `MessageHandlerOptions` as optional:

```typescript
export interface MessageHandlerOptions {
  exchange?: string;
  routingKey?: string | string[];

 // rest of the interface
}
```

I just pulled down a fresh copy of the repository and made the change myself to make sure and it works as intended. All of the decorators no longer require that you specify `exchange` or `routingKey` as long as you make the change that I've shown.

You need to actually build the project in order for the changes to be picked up in the e2e test app though. Either run `yarn build` after you make a change or just use `yarn build:watch` to be able to have everything recompile on the fly while you make changes